be/src/exprs/agg/aggregate.h

```diff
@@ -100,6 +106,13 @@ class AggregateFunction {
                                            int64_t peer_group_start, int64_t peer_group_end, int64_t frame_start,
                                            int64_t frame_end) const {}
 
+    // For window functions
+    // A peer group is all of the rows that are peers within the specified ordering.
+    // Rows are peers if they compare equal to each other using the specified ordering expression.
+    // Update batch single state with null
+    virtual void update_batch_single_state_null(FunctionContext* ctx, AggDataPtr __restrict state,
```

Comment:
Why ignore frame_start and frame_end?

Reply:
Current, if all not null, frame_end - frame_start may > 1. if has null, frame_end = frame_start + 1. 

https://github.com/StarRocks/starrocks/blob/03d261562a93ae35c1abd64cf76fac43c26e0a75/be/src/exprs/agg/nullable_aggregate.h#L415

so it will be only have one index if null

In the past, this interface was not well designed, and the batch interface should not be used for a single line. So this hidden semantics is still used here.

Reply:
modify to update_single_state_null